add search project id function

letusfly85 opened the pull request:
I want to get GitLab repo's projectId below.

https://github.com/letusfly85/drone/blob/master/plugin/remote/gitlab/gitlab.go#L165

Bugagazavr commented on deploy_keys.go:
```diff
@@ -47,6 +49,41 @@ Parameters:
     key_id The ID of a key
 
 */
+func (g *Gitlab) SearchProjectId(namespace string, name string) (id int, err error) {
```
Please, update godoc comments before this method or remove comments

Bugagazavr commented on deploy_keys.go:
```diff
@@ -10,6 +10,8 @@ const (
 	// PROJECT ID AND KEY ID
 	project_url_deploy_key = "/projects/:id/keys/:key_id" // Get single project deploy key
+
+	project_url_search_ids = "/projects/search/:query"
```
i think this should be moved to projects.go too

letusfly85 commented:
Thank you.

I moved function to projects.go,
and added comments.

letusfly85 commented:
Thank you!

I changed const from 'project_url_search_ids' to 'projects_search_url' in projects.go.
And I removed 'project_url_search_ids' from deploy_key.go.

letusfly85 commented:
Modified and I write a test code.

Please check.

Bugagazavr commented:
thanks

letusfly85 commented:
Thank you!
